projects/spectator/src/lib/internals/event-objects.ts

```diff
@@ -36,25 +38,50 @@ export function createTouchEvent(type: string, pageX: number = 0, pageY: number
 
 /** Dispatches a keydown event from an element. */
 export function createKeyboardEvent(type: string, keyOrKeyCode: string | number, target?: Element): KeyboardEvent {
-  const key = typeof keyOrKeyCode === 'string' && keyOrKeyCode;
-  const keyCode = typeof keyOrKeyCode === 'number' && keyOrKeyCode;
+  const { key, keyCode, modifiers } = parseKeyOptions(keyOrKeyCode);
 
   const event = document.createEvent('KeyboardEvent') as any;
   const originalPreventDefault = event.preventDefault;
 
   // Firefox does not support `initKeyboardEvent`, but supports `initKeyEvent`.
   if (event.initKeyEvent) {
```

Comment:
Angular material made updates to this code block, so this is the updated code from material.